File: ruby/rna-transcription/_iterations/1/rna_transcription.rb
module Complement
  NUCLEOTIDES = {
    "G" => "C",
    "C" => "G",
    "T" => "A",
    "A" => "U"
  }

  def self.of_dna(dna)
    dna.chars.map { |nucleotide| NUCLEOTIDES[nucleotide] }.join
  end
end

File: typescript/rna-transcription/_iterations/1/rna-transcription.ts
const dnaToRna: Record<string, string> = {
  'G': 'C',
  'C': 'G',
  'T': 'A',
  'A': 'U',
};

function toComplement(nucleotide: string): string {
  const complement = dnaToRna[nucleotide];
  if (!complement) {
    throw new Error('Invalid input DNA.');
  }
  return complement;
}

export function toRna(dna: string): string {
  return dna.split('').map((nucleotide) => toComplement(nucleotide)).join('');
}
